go/pkg/filemetadata/cache.go

```diff
@@ -17,14 +17,31 @@ func ResetGlobalCache() {
 
 // Cache is a store for file digests that supports invalidation.
 type fmCache struct {
-	Backend     *cache.SingleFlight
-	cacheHits   uint64
-	cacheMisses uint64
+	Backend        *cache.SingleFlight
+	cacheHits      uint64
+	cacheMisses    uint64
+	withDirContent bool
+}
+
+type Option func(*fmCache)
+
+// WithDirContent adds the list of files under a directory to the metadata.
+// exclude is currently ignored, but is meant to avoid caching content
+// for directories that are known to change their content during the lifetime
+// of the cache.
+func WithDirContent(exclude string) Option {
```

Comment:
can we introduce this option when we actually use it?

Reply:
It's needed to allow reclient to selectively toggle it via a flag. I'll remove the string argument for now and add a note about this being experimental.

Reply:
yeah I meant the exclude string arg!